### PR TITLE
Price Bound PSM

### PR DESCRIPTION
**Pull request by ElliotFriedman**
Price Bound PSM contract imported from fei. This contract will go live and have all PCV Fei needed inside of it so that users are free to redeem their Volt at any time.

A few changes had to be made to support the Volt system:
- Calling volt() instead of fei() to fetch the contract balance
- inverting the price in the deploy script to properly price the asset in the PSM
- formatting by a linter
- Surplus target is 10b Fei so that it will not deposit Fei into the PCV Deposit for quite a few years
- Target PCV deposit is the Volt fuse PCV Deposit which will not do anything with received Fei

This contract can't mint Volt as it doesn't have a reference to the GlobalRateLimitedMinter, and we will not give this contract the minter role. Instead, this contract will have Volt minted into it which will save users gas when they wish to purchase Volt.

diff: https://www.diffchecker.com/SfSE30vu

### CODE REVIEW

**Review by eswak — COMMENTED** [review]
Code looks good, but do you have a way to test after deployment ? I'd be more comfortable if I saw a script to do 1 mint and 1 redeem call on a local fork and check the balance before/after each

File: scripts/priceBoundPSMDeploy.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  FEI,
+  VOLT,
+  ORACLE_PASS_THROUGH_ADDRESS,
+  VOLT_FUSE_PCV_DEPOSIT,
+  /// fees
+  MINT_FEE_BASIS_POINTS,
+  REDEEM_FEE_BASIS_POINTS
+} = config;
+
+const feiReservesThreshold = ethers.utils.parseEther('10000000000');
```
Comment (thomas-waite) [inline scripts/priceBoundPSMDeploy.ts:25]: Does this not need multiplying by the number of decimals in Fei? When we set the reserves threshold for the ethPSM it was multiplied by 18 decimal places

**Review by thomas-waite — COMMENTED** [review]
Looks good to me, echo Erwan's point about how either now or in the near future you're going to need a way to simulate that these deployments work on a forked version of mainnet

**Review by k-xo — APPROVED** [review]
LGTM!

**Comment by ElliotFriedman** [comment]
> Code looks good, but do you have a way to test after deployment ? I'd be more comfortable if I saw a script to do 1 mint and 1 redeem call on a local fork and check the balance before/after each

Added integration tests in to demonstrate functionality on mainnet!

**Comment by ElliotFriedman** [comment]
Price bound PSM is used so that if the oracle sends in invalid data, the system shuts down and doesn't allow trading at an incorrect price. This will mean that every year we will have to change the floor and ceiling prices to account for this.

**Pull request merged**